[firebase_storage] Support for download progress

awazgyawali opened the pull request:
## Description

Adds support for Download Progress;

## Related Issues

https://github.com/FirebaseExtended/flutterfire/issues/420

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

pc-brainants commented:
@googlebot I consent.

awazgyawali commented:
For those who might be wondering on how to use this right away, use this on your pub spec.
```  
  firebase_storage:
    git:
      url: https://github.com/awazgyawali/flutterfire
      path: packages/firebase_storage
      ref: storage_download_progress
```

sudhanshu16 commented:
Hey @awazgyawali  
Correct me if I am wrong, there is no progress callback in this implementation but there are booleans to check if the task is in progress, right?
If so, then do you plan on implementing a callback mechanism? Basically a `function (int progress, int total)` that could be called on each progress event?

awazgyawali commented:
There is support for progress. Its very similar to the upload one.
```
    StorageFileDownloadTask task =
        _storage.ref().child(attachment.storageReference).writeToFile(file);
    task.events.listen((event) {
      double progress =
          event.snapshot.bytesTransferred / event.snapshot.totalByteCount;
          // TODO: This is progress
    });
    await task.onComplete; // This is resolved once download is completed
```

sudhanshu16 commented:
@awazgyawali Lol I figured it out minutes after I commented so I deleted it. Thanks anyways

yanivshaked commented:
Any timeline on when will this because a part of the official release?

awazgyawali commented:
There is an ongoing refactor of the whole package paradigm from invertase at #2582 

Once the is stable, I will maintain it as the repo and might get reviewed. Please expect a timeline of around about a month.

nstosic-go commented:
Thanks for the update @awazgyawali. Just to confirm, you're keeping this PR on hold until the `firebase_storage` package rewrite has been merged. You're not going to wait for **all** FlutterFire packages to be rewritten, right?

awazgyawali commented:
Yes, we are only interested on `firebase_storage` change however the team might not have time to review or approve the changes until all the packages are refactored. I'll do my part anyway to complete it once the rework of  storage package is done but we might have to wait till everything closes out.

yanivshaked commented:
> For those who might be wondering how to use this right away, use this on your pub spec.
> 
> ```
>   firebase_storage:
>     git:
>       url: https://github.com/awazgyawali/flutterfire
>       path: packages/firebase_storage
>       ref: storage_download_progress
> ```
> 
> Note: On the Windows environment, you might get weird long pathname issues, follow [here](https://stackoverflow.com/a/52699496/3931968).
> 
> A star on my fork would be appreciated if it helped you, Thanks. :-)

I have tried using the above suggestion, but got the following errors (on firebase_auth):
```

C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java:8: error: package android.support.annotation does not exist
import android.support.annotation.NonNull;
                                 ^
C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java:538: error: cannot find symbol
    public void onComplete(@NonNull Task<AuthResult> task) {
                            ^
  symbol:   class NonNull
  location: class FirebaseAuthPlugin.SignInCompleteListener
C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java:558: error: cannot find symbol
    public void onComplete(@NonNull Task<Void> task) {
                            ^
  symbol:   class NonNull
  location: class FirebaseAuthPlugin.TaskVoidCompleteListener
C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java:576: error: cannot find symbol
    public void onComplete(@NonNull Task<ProviderQueryResult> task) {
                            ^
  symbol:   class NonNull
  location: class FirebaseAuthPlugin.ProvidersCompleteListener
C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java:175: error: cannot find symbol
                      public void onComplete(@NonNull Task<AuthResult> task) {
                                              ^
  symbol: class NonNull
C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java:270: error: cannot find symbol
          public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
                                          ^
  symbol: class NonNull
C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java:435: error: cannot find symbol
              public void onComplete(@NonNull Task<GetTokenResult> task) {
                                      ^
  symbol: class NonNull
C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java:487: error: cannot find symbol
          public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
                                          ^
  symbol: class NonNull
Note: C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
8 errors

```
FAILURE: Build failed with an exception.

Is there any way to use a version of firebase_auth that is aligned with the changes in the above suggested repository?

Thanks!

awazgyawali commented:
Use the following
```
  firebase_auth:
    git:
      url: https://github.com/awazgyawali/flutterfire
      path: packages/firebase_auth/firebase_auth
      ref: 5ac07af
```

yanivshaked commented:
> firebase_storage:
>     git:
>       url: https://github.com/awazgyawali/flutterfire
>       path: packages/firebase_auth/firebase_storage
>       ref: 5ac07af

Tried, got:

```
PS C:\dev\12Tech12\git\client\tw_app> flutter pub get
Could not find a file named "packages/firebase_auth/firebase_storage/pubspec.yaml" in https://github.com/awazgyawali/flutterfire 5ac07af765bf43974d30f654dd84fe5417259787.
Running "flutter pub get" in tw_app...
pub get failed (1; Could not find a file named "packages/firebase_auth/firebase_storage/pubspec.yaml" in https://github.com/awazgyawali/flutterfire 5ac07af765bf43974d30f654dd84fe5417259787.)
```

awazgyawali commented:
I edited my comment :-)

yanivshaked commented:
> I edited my comment :-)

Still getting the following errors:
```

C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java:8: error: package android.support.annotation does not exist
import android.support.annotation.NonNull;
                                 ^
C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java:538: error: cannot find symbol
    public void onComplete(@NonNull Task<AuthResult> task) {
                            ^
  symbol:   class NonNull
  location: class FirebaseAuthPlugin.SignInCompleteListener     
C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java:558: error: cannot find symbol
    public void onComplete(@NonNull Task<Void> task) {
                            ^
  symbol:   class NonNull
  location: class FirebaseAuthPlugin.TaskVoidCompleteListener
C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java:576: error: cannot find symbol
    public void onComplete(@NonNull Task<ProviderQueryResult> task) {   
                            ^
  symbol:   class NonNull
  location: class FirebaseAuthPlugin.ProvidersCompleteListener
C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java:175: error: cannot find symbol
                      public void onComplete(@NonNull Task<AuthResult> task) {
                                              ^
  symbol: class NonNull
C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java:270: error: cannot find symbol
          public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
                                          ^
  symbol: class NonNull
C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java:435: error: cannot find symbol
              public void onComplete(@NonNull Task<GetTokenResult> task) {
                                      ^
  symbol: class NonNull
C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java:487: error: cannot find symbol
          public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
                                          ^
  symbol: class NonNull
Note: C:\Users\Yaniv\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\firebase_auth-0.6.2+1\android\src\main\java\io\flutter\plugins\firebaseauth\FirebaseAuthPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
8 errors

FAILURE: Build failed with an exception.
```

Which version of firebase_auth should I use? (currently using latest release)

yanivshaked commented:
I could not establish a working set of firebase_auth, firebase_storage, firebase_messaging and firebase_core with the support for download progress. 
If possible, please provide the correct sources to use in pubspec.yaml for all 4 packages.

Thanks!

awazgyawali commented:
[pubspec.txt](https://github.com/FirebaseExtended/flutterfire/files/4986208/pubspec.txt)
Its working for me, please filter your required versions and package

yanivshaked commented:
> [pubspec.txt](https://github.com/FirebaseExtended/flutterfire/files/4986208/pubspec.txt)
> Its working for me, please filter your required versions and package

Thanks, following set seems to be working:
```

  # Firebase Core Flutter SDK
  firebase_core: ^0.4.5
  # Flutter plugin for Firebase Auth, enabling Android and iOS authentication using passwords, phone numbers and identity providers like Google, Facebook and Twitter.
  firebase_auth: ^0.16.1
  # Flutter plugin for Firebase Cloud Storage, a powerful, simple, and cost-effective object storage service for Android and iOS.
  firebase_storage:
    git:
      url: https://github.com/awazgyawali/flutterfire
      path: packages/firebase_storage
      ref: 5ac07af
  # Flutter plugin for Firebase Cloud Messaging, a cross-platform messaging solution that lets you reliably deliver messages on Android and iOS.
  firebase_messaging: ^6.0.16

```

yanivshaked commented:
Hi,

I have noticed the following incorrect behavior when using pause/resume functionality on download operation. There is a glitch in the progress reporting due to:
- incorrect totalBytesCount reported after resume operation (only the remainder size is reported).
- incorrect bytesTransferred reported shortly after the resume operation (the correct size is back after a few progress events)

See following log to clarify the issue, let me know if you need more information:
```

I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.19667280368881546 = 14417920 / 73309170
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.20024867284679393 = 14680064 / 73309170
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.2038245420047724 = 14942208 / 73309170
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.20740041116275085 = 15204352 / 73309170
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.21097628032072932 = 15466496 / 73309170
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.2145521494787078 = 15728640 / 73309170
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.21812801863668624 = 15990784 / 73309170
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.2217038877946647 = 16252928 / 73309170
I/flutter (13637): calling downloadWrapper.storageFileDownloadTask.pause
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.2217038877946647 = 16252928 / 73309170
W/StorageTask(13637): unable to change internal state to: INTERNAL_STATE_IN_PROGRESS isUser: false from state:INTERNAL_STATE_PAUSING
I/flutter (13637): transferEventStream::Pause
I/flutter (13637): pause
I/flutter (13637): 2020-08-01 05:04:03.544983Z General, D: ChatViewItem: got streamed feed
I/flutter (13637): 2020-08-01 05:04:03.547224Z General, D: ChatPage: got streamed feed: 1691, 35, 131, 2020-07-31 05:51:50.299Z
I/flutter (13637): calling downloadWrapper.storageFileDownloadTask.resume
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.4505595139052863 = 33030144 / 73309170
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.4505595139052863 = 33030144 / 73309170
D/FileDownloadTask(13637): Resuming download file /data/user/0/com.twtech.twelve/app_flutter/videos/2020_08_01T05_03_54_078542659fb1c0-d3b4-11ea-e81e-a3ba6807d756 at 16515072
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.2954042161211892 = 16777216 / 56794098
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.3000199069980828 = 17039360 / 56794098
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.30463559787497635 = 17301504 / 56794098
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.30925128875186997 = 17563648 / 56794098
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.31386697962876353 = 17825792 / 56794098
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.3184826705056571 = 18087936 / 56794098
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.3230983613825507 = 18350080 / 56794098
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.3277140522594443 = 18612224 / 56794098
I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.33232974313633784 = 18874368 / 56794098
```

yanivshaked commented:
When trying to pause the transfer on iOS, the operation fails with the following message:
```
[VERBOSE-2:ui_dart_state.cc(157)] Unhandled Exception: PlatformException(pause_error, task == null, null)
#0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:569)
#1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:156)
<asynchronous suspension>
```
Is there some different setup required for iOS?
(Same codebase on Android is working)

yanivshaked commented:
@awazgyawali Appreciate your help with the above two issue.

yanivshaked commented:
Dear @awazgyawali ,
I have just updated to the latest firebase_storage version (5.0.0-dev.1).
The issue I have reported below (on Aug 1st) still exists.

Is there any chance someone can attend it?
Otherwise, if you can provide me with some guidelines, I will try to fix it and submit a PR.

Thanks,
Yaniv

> Hi,
> 
> I have noticed the following incorrect behavior when using pause/resume functionality on download operation. There is a glitch in the progress reporting due to:
> 
> * incorrect totalBytesCount reported after resume operation (only the remainder size is reported).
> * incorrect bytesTransferred reported shortly after the resume operation (the correct size is back after a few progress events)
> 
> See following log to clarify the issue, let me know if you need more information:
> 
> ```
> 
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.19667280368881546 = 14417920 / 73309170
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.20024867284679393 = 14680064 / 73309170
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.2038245420047724 = 14942208 / 73309170
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.20740041116275085 = 15204352 / 73309170
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.21097628032072932 = 15466496 / 73309170
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.2145521494787078 = 15728640 / 73309170
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.21812801863668624 = 15990784 / 73309170
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.2217038877946647 = 16252928 / 73309170
> I/flutter (13637): calling downloadWrapper.storageFileDownloadTask.pause
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.2217038877946647 = 16252928 / 73309170
> W/StorageTask(13637): unable to change internal state to: INTERNAL_STATE_IN_PROGRESS isUser: false from state:INTERNAL_STATE_PAUSING
> I/flutter (13637): transferEventStream::Pause
> I/flutter (13637): pause
> I/flutter (13637): 2020-08-01 05:04:03.544983Z General, D: ChatViewItem: got streamed feed
> I/flutter (13637): 2020-08-01 05:04:03.547224Z General, D: ChatPage: got streamed feed: 1691, 35, 131, 2020-07-31 05:51:50.299Z
> I/flutter (13637): calling downloadWrapper.storageFileDownloadTask.resume
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.4505595139052863 = 33030144 / 73309170
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.4505595139052863 = 33030144 / 73309170
> D/FileDownloadTask(13637): Resuming download file /data/user/0/<CUT_package_name>/app_flutter/videos/2020_08_01T05_03_54_078542659fb1c0-d3b4-11ea-e81e-a3ba6807d756 at 16515072
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.2954042161211892 = 16777216 / 56794098
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.3000199069980828 = 17039360 / 56794098
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.30463559787497635 = 17301504 / 56794098
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.30925128875186997 = 17563648 / 56794098
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.31386697962876353 = 17825792 / 56794098
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.3184826705056571 = 18087936 / 56794098
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.3230983613825507 = 18350080 / 56794098
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.3277140522594443 = 18612224 / 56794098
> I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.33232974313633784 = 18874368 / 56794098
> ```

awazgyawali commented:
> 
> Dear @awazgyawali ,
> I have just updated to the latest firebase_storage version (5.0.0-dev.1).
> The issue I have reported below (on Aug 1st) still exists.
> 
> Is there any chance someone can attend it?
> Otherwise, if you can provide me with some guidelines, I will try to fix it and submit a PR.
> 
> Thanks,
> Yaniv
> 
> > Hi,
> > 
> > I have noticed the following incorrect behavior when using pause/resume functionality on download operation. There is a glitch in the progress reporting due to:
> > 
> > * incorrect totalBytesCount reported after resume operation (only the remainder size is reported).
> > * incorrect bytesTransferred reported shortly after the resume operation (the correct size is back after a few progress events)
> > 
> > See following log to clarify the issue, let me know if you need more information:
> > 
> > ```
> > 
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.19667280368881546 = 14417920 / 73309170
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.20024867284679393 = 14680064 / 73309170
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.2038245420047724 = 14942208 / 73309170
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.20740041116275085 = 15204352 / 73309170
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.21097628032072932 = 15466496 / 73309170
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.2145521494787078 = 15728640 / 73309170
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.21812801863668624 = 15990784 / 73309170
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.2217038877946647 = 16252928 / 73309170
> > I/flutter (13637): calling downloadWrapper.storageFileDownloadTask.pause
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.2217038877946647 = 16252928 / 73309170
> > W/StorageTask(13637): unable to change internal state to: INTERNAL_STATE_IN_PROGRESS isUser: false from state:INTERNAL_STATE_PAUSING
> > I/flutter (13637): transferEventStream::Pause
> > I/flutter (13637): pause
> > I/flutter (13637): 2020-08-01 05:04:03.544983Z General, D: ChatViewItem: got streamed feed
> > I/flutter (13637): 2020-08-01 05:04:03.547224Z General, D: ChatPage: got streamed feed: 1691, 35, 131, 2020-07-31 05:51:50.299Z
> > I/flutter (13637): calling downloadWrapper.storageFileDownloadTask.resume
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.4505595139052863 = 33030144 / 73309170
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.4505595139052863 = 33030144 / 73309170
> > D/FileDownloadTask(13637): Resuming download file /data/user/0/<CUT_package_name>/app_flutter/videos/2020_08_01T05_03_54_078542659fb1c0-d3b4-11ea-e81e-a3ba6807d756 at 16515072
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.2954042161211892 = 16777216 / 56794098
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.3000199069980828 = 17039360 / 56794098
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.30463559787497635 = 17301504 / 56794098
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.30925128875186997 = 17563648 / 56794098
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.31386697962876353 = 17825792 / 56794098
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.3184826705056571 = 18087936 / 56794098
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.3230983613825507 = 18350080 / 56794098
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.3277140522594443 = 18612224 / 56794098
> > I/flutter (13637): _progress = event.bytesTransferred / event.totalByteCount => 0.33232974313633784 = 18874368 / 56794098
> > ```
> 
> 

@yanivshaked This PR has not been merged, @Salakar  and the team re worked on the plugin and introduced the download progress feature. 

May be he can help you with this.

yanivshaked commented:
Dear @Salakar,
Maybe you can help with this issue, I have opened a [bug report](https://github.com/FirebaseExtended/flutterfire/issues/3723).

Thanks,
Yaniv
